torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -1625,6 +1632,20 @@ void LLVMCodeGenImpl::visit(const Intrinsics* v) {
         throw unimplemented_lowering(v);
       } break;
     }
+  } else if (v->dtype().is_integral() && v->op_type() == kFabs) {
+    // abs is only intrinsic defined for integer inputs in pytorch eager
+    v->params().front()->accept(this);
+    if (is_unsigned_integral(v->dtype().scalar_type())) {
+      return;
+    }
+    // TODO: use llvm.abs intrinsic for LLVM 12
+    auto zero = llvm::ConstantInt::get(value_->getType(), 0);
+    auto neg_value = irb_.CreateSub(zero, value_);
+    auto icmp = irb_.CreateICmpSGT(value_, zero);
+    value_ = irb_.CreateSelect(icmp, value_, neg_value);
```

Comment:
this could be implemented differently - any tips on how I should be going abt profiling generating assembly vs other implementations / if it matters ?

Reply:
If you want to look at the generated assembly there's a DEBUG_PRINT constant you can define somewhere (I think in llvm_codegen or llvm_jit) and that'll dump the llvm ir and the x86 assembly.  Might be worth checking out.  Could also add an operator level benchmark in benchmarks/tensorexpr

Reply:
I looked up with the IR and it was reasonable. I guess just whether something like a shift and & would be faster. Will look into it more as a follow up with @navahgar when we get some op benchmarks going.